### PR TITLE
[Routing] always add `_route_path` as request attribute

### LINKED ISSUE #41426: Add a route pattern (route path) to the UrlMatcher/RouterListener
**Description**
Add a route path (URL pattern) to the returned URL matcher parameters. Currently, UrlMatcher returns an array (example from demo app):

```
[
  "_route" => "blog_index",
  "page" => "1",
  "_format" => "html",
  "_controller" => "App\Controller\BlogController::index",
  "_locale" => "fr"
]
```

But sometimes the route pattern is needed (like `/{_locale}/blog/`).

**Example**
The real use case is: my client wants the integration of a site monitoring system with Symfony, and wants to group information of each request by URL patterns. Currently, there's no other way to get route path pattern except using `Router::getRouteCollection()` which is not recommended. Also, `Router::match()` leads to Twig related exception in the demo app:
```
An exception has been thrown during the rendering of a template ("No routes found for "/_fragment/".").
```

To reproduce a Twig exception, you need to set up a demo app, create an event listener that is executed before RouterListener (e.g. priority 9999), inject Router, and execute `$this->router->match($request->getPathInfo());` on `kernel.request` event

### PR DESCRIPTION
**Pull request by cdaguerre**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix https://github.com/symfony/symfony/issues/41426
| License       | MIT
| Doc PR        | n.a

This PR adds the `_path` attribute to attributes returned by the request matcher.
If I'm not mistaken, this information is actually hard to get without using the `RouteCollection` at runtime or manipulating cache files directly (be they generated by the routing cache warmer or by a custom cache warmer).
I'b be glad to add/fix tests if there is an agreement on the principle ;)

### CODE REVIEW

**Review by GromNaN — CHANGES_REQUESTED** [review]
Hello @cdaguerre. Thanks opening this PR while [you explore alternative solutions](https://github.com/worldia/instrumentation-bundle/pull/12#issuecomment-1288716329).

Major APM (datadog, newrelic and instana) use the route name to name the span. This keep distinction between routes with the same path pattern but distinct host/condition/requirements...

File: src/Symfony/Component/Routing/Matcher/UrlMatcher.php [hunk 1]
```diff
@@ -201,6 +201,7 @@ protected function getAttributes(Route $route, string $name, array $attributes):
             unset($defaults['_canonical_route']);
         }
         $attributes['_route'] = $name;
+        $attributes['_path'] = $route->getPath();
```
Comment (GromNaN) [inline src/Symfony/Component/Routing/Matcher/UrlMatcher.php]: The name should be `_route_path` to avoid confusion with `Request::getPath()` and the `_path` query parameter used for fragments.

Comment (cdaguerre) [reply src/Symfony/Component/Routing/Matcher/UrlMatcher.php]: done

**Comment by cdaguerre** [comment]
> Major APM (datadog, newrelic and instana) use the route name to name the span. This keep distinction between routes with the same path pattern but distinct host/condition/requirements...

That is indeed one of the alternatives. However, it is not really compliant with [OpenTelemetry's semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server)...
Disambiguating/distinguishing host etc. is done with other attributes like `http.scheme`, `net.host.name`.

File: src/Symfony/Component/Routing/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ CHANGELOG
 ---
 
  * Add `Requirement::POSITIVE_INT` for common ids and pagination
+ * The request matcher now adds the path of the matched route as a `_path` attribute
```
Comment (GromNaN) [inline src/Symfony/Component/Routing/CHANGELOG.md]:
```suggestion
 * The request matcher now adds the path of the matched route as a `_route_path` attribute
```

File: src/Symfony/Component/Routing/Matcher/Dumper/CompiledUrlMatcherDumper.php [hunk 3]
```diff
@@ -398,7 +398,7 @@ private function compileStaticPrefixCollection(StaticPrefixCollection $tree, \st
             }
 
             $state->mark += 3 + $state->markTail + \strlen($regex) - $prefixLen;
-            $state->markTail = 2 + \strlen($state->mark);
+            $state->markTail = 2 + \strlen((string) $state->mark);
```
Comment (GromNaN) [inline src/Symfony/Component/Routing/Matcher/Dumper/CompiledUrlMatcherDumper.php]: This seems to be not related to your PR. Did you encountered an issue without explicit cast?

**Review by GromNaN — CHANGES_REQUESTED** [review]

**Comment by cdaguerre** [comment]
I made the changes you requested and fixed the tests.
Failing tests on 8.1 seem unrelated to these changes.
However, I need your advice on fixing the high/low-deps tests, because these tests succeed in 8.1 and 8.2 and I'm not sure what I should do here.

**Comment by Tobion** [comment]
I can see that having the route path available might be useful. But it seems arbitrary to me that the returned values include the route path but none of the other route information like route host, scheme etc.
So this does not seem like the right solution to me.

**Review by GromNaN — CHANGES_REQUESTED** [review]
Same opinion. Getting raw route config from the requests is only a partial solution for your need. 

There is a risk that these values will be misused to generate urls.

**Comment by GromNaN** [comment]
Hi @cdaguerre, I closed your MR because we think this informations should not be exposed as this. I see you found an alternative in https://github.com/worldia/instrumentation-bundle/pull/12 that is good and allows more specific code.

**Comment by cdaguerre** [comment]
alright, thanks
